src/test/java/seedu/address/model/leave/LeaveTest.java

```diff
@@ -0,0 +1,128 @@
+package seedu.address.model.leave;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalLeaves.ALICE_LEAVE;
+import static seedu.address.testutil.TypicalLeaves.BOB_LEAVE;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BOB;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.leave.exceptions.EndBeforeStartException;
+import seedu.address.testutil.LeaveBuilder;
+
+public class LeaveTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Leave(null, null, null, null));
```

Comment:
Can include tests where only one of the fields is null, to ensure that all fields have to be non-null.